mirrord-layer/src/pod_api.rs

```diff
@@ -255,21 +253,21 @@ async fn create_ephemeral_container_agent(
             "privileged": true,
         },
         "imagePullPolicy": config.image_pull_policy,
-        "targetContainerName": config.impersonated_container_name,
+        "targetContainerName": pod_target.container_name,
         "env": [{"name": "RUST_LOG", "value": config.agent_rust_log}],
         "command": agent_command_line,
     }))?;
     debug!("Requesting ephemeral_containers_subresource");
 
     let mut ephemeral_containers_subresource = pods_api
-        .get_subresource("ephemeralcontainers", &config.impersonated_pod_name)
+        .get_subresource("ephemeralcontainers", &pod_target.pod_name)
```

Comment:
The `pod_target` will be consumed in this function, so why not do a de-structuring assignment to reduce the amount of `pod_target.`? Do it like so:

```rust
let PodTarget { pod_name, container_name } = pod_target;
```

You could go one step further and be even fancier by doing this at the function parameter:

```rust
async fn create_ephemeral_container_agent(PodData { pod_name, container_name }: PodData)
```

Up to you right now (and maybe open a discussion around this style to let our pals vote).